servicebox-app/src/main/java/io/halkyon/resource/page/ServiceResource.java

```
@@ -83,7 +84,11 @@ private TemplateInstance showList(List<Service> services) {
     @Consumes(MediaType.APPLICATION_JSON)
     @Path("/name/{name}")
     public io.halkyon.model.Service findByName(@PathParam("name") String name) {
-        return io.halkyon.model.Service.findByName(name);
+        Service service = Service.findByName(name);
+        if (service == null) {
+            throw new WebApplicationException("Service with name " + name + " does not exist.", 404);
```

we should throw the NotFoundException exception instead.